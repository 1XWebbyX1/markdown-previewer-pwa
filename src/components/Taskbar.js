import React from 'react'
//import '@fortawesome/fontawesome-free/css/all.css'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faSave, faInfoCircle, faBold, faItalic, faQuoteLeft, faLink, faImage, faList, faListOl, faCode, faToggleOn} from '@fortawesome/free-solid-svg-icons'


class Taskbar extends React.Component{
  render(){
    return(
      <div id='taskbar'>
      <ul>
       <li><div onClick={this.props.save} className='fa fa-save'><FontAwesomeIcon id="save" className='fa fa-save icon' icon={faSave}/></div></li>
       <li><div onClick={this.props.toggleInfo} className='fa fa-info-circle'><FontAwesomeIcon id="info" className='fa fa-info-circle icon' icon={faInfoCircle}/></div></li>
       <li><div onClick={this.props.onClick} className='fa fa-bold'><FontAwesomeIcon id="bold" className='fa fa-bold icon' icon={faBold}/></div></li>
       <li><div onClick={this.props.onClick} className='fa fa-italic'><FontAwesomeIcon id="italic" className='fa fa-italic icon' icon={faItalic}/></div></li>
       <li><div onClick={this.props.onClick} className='fa fa-quote-left'><FontAwesomeIcon id="quote" className='fa fa-quote-left icon' icon={faQuoteLeft}/></div></li>
       <li><div onClick={this.props.onClick} className='fa fa-link'><FontAwesomeIcon id="link" className='fa fa-link icon' icon={faLink}/></div></li>
       <li><div onClick={this.props.onClick} className='fa fa-image'><FontAwesomeIcon id="picture" className='fa fa-image icon' icon={faImage}/></div></li>
       <li><div onClick={this.props.onClick} className='fa fa-list'><FontAwesomeIcon className='fa fa-list icon' icon={faList}/></div></li>
       <li><div onClick={this.props.onClick} className='fa fa-list-ol'><FontAwesomeIcon className='fa fa-list-ol icon' icon={faListOl}/></div></li>
       <li><div onClick={this.props.onClick} className='fa fa-code'><FontAwesomeIcon className='fa fa-code icon' icon={faCode}/></div></li>
       <li><div onClick={this.props.switchTheme} className='fa fa-toggle-on'><FontAwesomeIcon id="theme-switch" className='fa fa-toggle-on icon' icon={faToggleOn}/></div></li>
       </ul>
      </div>

    )
  }
}

export default Taskbar;
