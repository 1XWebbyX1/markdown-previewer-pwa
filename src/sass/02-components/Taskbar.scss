#taskbar{
  position: fixed;
  height: auto;
  width: 100%;
  margin: 0;
  background: var(--toolbar);
  color: var(--textFont);
  border-bottom: .5px solid black;

  >ul{
    display: flex;
    justify-content: center;
    list-style-type: none;
    text-decoration: none;

  li {
    height: 100%;
    border-right: 0.5px solid black;
    padding-right: 2.5%;
    margin-right: 2.5%;
  i {
    z-index: 99;
    font-size: calc(0.9em + 1vmax);
    cursor: pointer;
    &:hover {
      transform: scale(1.2);
    }
   }
  }
 }
}
