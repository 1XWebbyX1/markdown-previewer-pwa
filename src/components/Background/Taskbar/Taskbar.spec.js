// External Depedencies
import React from 'react';
import { shallow, mount, configure } from 'enzyme';
// Our Dependencies
import { expect } from '../../utils/chai';
import Adapter from 'enzyme-adapter-react-16';
import renderer from "react-test-renderer";
// Our Component
import Taskbar from './Taskbar';


configure({ adapter: new Adapter() });

describe('Taskbar', () => {
  const mockOnClickFunc = jest.fn();
  const mockToggleInfoFunc = jest.fn();
  const mockSaveFunc = jest.fn();
  const mockSwitchThemeFunc = jest.fn();

  //to make sure all list items are rendered
  it("renders correctly", () => {
     const output = renderer.create(
         <Taskbar onClick={mockOnClickFunc} toggleInfo={mockToggleInfoFunc} switchTheme={mockSwitchThemeFunc} save={mockSaveFunc} />
   ).toJSON();
     expect(output).to.matchSnapshot();
  });

  it('renders 11 list elements', () => {
    let app = shallow(
<Taskbar onClick={mockOnClickFunc} toggleInfo={mockToggleInfoFunc} switchTheme={mockSwitchThemeFunc} save={mockSaveFunc} />     );
    expect(app.find('li').length).to.equal(11);
    app.unmount();
  });

});
