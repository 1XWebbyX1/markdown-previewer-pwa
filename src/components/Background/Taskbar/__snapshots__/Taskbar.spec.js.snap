// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Taskbar renders correctly 1`] = `
<div
  id="taskbar"
>
  <ul>
    <li>
      <i
        className="fa fa-save"
        id="save"
        onClick={[MockFunction]}
      />
    </li>
    <li>
      <i
        className="fa fa-info-circle"
        id="info"
        onClick={[MockFunction]}
      />
    </li>
    <li>
      <i
        className="fa fa-bold"
        id="bold"
        onClick={[MockFunction]}
      />
    </li>
    <li>
      <i
        className="fa fa-italic"
        id="italic"
        onClick={[MockFunction]}
      />
    </li>
    <li>
      <i
        className="fa fa-quote-left"
        id="quote"
        onClick={[MockFunction]}
      />
    </li>
    <li>
      <i
        className="fa fa-link"
        id="link"
        onClick={[MockFunction]}
      />
    </li>
    <li>
      <i
        className="fa fa-image"
        id="picture"
        onClick={[MockFunction]}
      />
    </li>
    <li>
      <i
        className="fa fa-list"
        onClick={[MockFunction]}
      />
    </li>
    <li>
      <i
        className="fa fa-list-ol"
        onClick={[MockFunction]}
      />
    </li>
    <li>
      <i
        className="fa fa-code"
        onClick={[MockFunction]}
      />
    </li>
    <li>
      <i
        className="fa fa-toggle-on"
        id="theme-switch"
        onClick={[MockFunction]}
      />
    </li>
  </ul>
</div>
`;
